From fd02387cc1c7efacd5d48d18a9d1d6b7c715c7ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 5 Jul 2016 17:27:31 +0300
Subject: [PATCH 2/6] abuild: fix cross compilation undeps()

"apk del" now returns error if the package name does not exist.
Thus when cross compiling, always install the virtual
.makedepends-$pkgname name for chroot also since undeps() will
try to uninstall it always.
---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index db82cfd..883c661 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1812,7 +1812,7 @@ builddeps() {
 			--simulate --quiet $builddeps || return 1
 		$SUDO_APK add --repository "$abuildrepo" $apk_opt_wait \
 			--virtual .makedepends-$pkgname $builddeps || return 1
-		if [ -n "$hostdeps" ]; then
+		if cross_compiling; then
 			$SUDO_APK add --root "$CBUILDROOT" --repository "$abuildrepo" $apk_opt_wait \
 				--simulate --quiet $hostdeps || return 1
 			$SUDO_APK add --root "$CBUILDROOT" --repository "$abuildrepo" $apk_opt_wait \
-- 
2.9.0

